adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -839,8 +842,9 @@ def _wait_for_msg(self, timeout=0.1):
             pid = pid[0] << 0x08 | pid[1]
             sz -= 0x02
         # read message contents
-        msg = self._sock_exact_recv(sz)
-        self._handle_on_message(self, topic, str(msg, "utf-8"))
+        raw_msg = self._sock_exact_recv(sz)
```

Comment:
Could we automatically detect here somehow, instead?

Reply:
Not sure how to detect if the `bytearray` content is a Unicode string or a binary blob 😞 . 

We can add a try/except block and convert it to a Unicode string, if it fails it returns the message in binary, but not sure if is consistent behavior because users might have to check the data type on the callback function.

Reply:
Yup, let's have the user handle it in user code instead.